### ISSUE TITLE
Forbid usage of analysis-api annotations that indicate code designed for internal use only

### ISSUE THREAD

**Issue by 3flex** [detekt/detekt#8438]
## Expected Behavior
detekt doesn't use any analysis-api API that requires the use of one of these annotations.

## Current Behavior
Can only catch this in PR code reviews.

## Context
Several opt ins are used in analysis-api that indicate code should not be used externally, see https://github.com/JetBrains/kotlin/blob/v2.2.20-Beta2/analysis/analysis-api/src/org/jetbrains/kotlin/analysis/api/annnotations.kt

Examples:
* KaImplementationDetail
* KaNonPublicApi
* KaIdeApi
* KaPlatformInterface

The final one in the linked file, `KaExperimentalApi`, is OK as the code is more likely to be stabilised than not in future.

A new "ForbiddenOptIn" rule might be useful, though most usage will be caught by adding to the list of forbidden imports.

**Comment by BraisGabin** [comment]
We have the new forbidden annotation rule. Maybe we could expand it to forbid with string literals. If that doesn't work +1 to forbid optins

**Comment by marschwar** [comment]
I don't think we can use the `ForbiddenAnnotation` rule. The annotations are configured by `ValueWithReason` and I do not see a clear way to add an additional list of values.

**Comment by BraisGabin** [comment]
My idea was more something like:

```yaml
- 'kotlin.OptIn(String = "KaImplementationDetail")' 
```

**Comment by 3flex** [comment]
Not sure that helps much when a class needs to be passed to the annotation like opt ins need.

Can open a new issue for designing a new rule or extending an existing one. For now I'll just forbid the imports and close this.

**Comment by 3flex** [comment]
Will do this after updating to Kotlin 2.3 so the opt outs and suppressions are most up to date.
